packages/vscode-extension/src/debug/commonUtils.ts

```diff
@@ -396,3 +397,18 @@ export async function loadTeamsFxDevScript(componentRoot: string): Promise<strin
     return undefined;
   }
 }
+
+// Helper functions for local debug correlation-id, only used for telemetry
+let localDebugCorrelationId: string | undefined = undefined;
+export function startLocalDebugSession(): string {
+  localDebugCorrelationId = uuid.v4();
+  return getLocalDebugSessionId();
+}
+
+export function endLocalDebugSession() {
+  localDebugCorrelationId = undefined;
+}
+
+export function getLocalDebugSessionId(): string {
+  return localDebugCorrelationId || "undefined";
```

Comment:
Use another string instead of "undefined", to distinguish with real undefined case. E.g., "local-debug-not-started".
(Just like do not set your display name to `null` :P )

Reply:
Or, "no-session-{$uuid}", if we do want to provide correlation for **all** scenarios.

Reply:
Good point, thanks!

Reply:
fixed